stardog/connection.py

```diff
@@ -122,17 +137,19 @@ def rollback(self):
         self.client.post("/transaction/rollback/{}".format(self.transaction))
         self.transaction = None
 
-    def commit(self):
+    def commit(self) -> CommitResult:
         """Commits the current transaction.
 
         Raises:
           stardog.exceptions.TransactionException
             If currently not in a transaction
         """
         self._assert_in_transaction()
-        self.client.post("/transaction/commit/{}".format(self.transaction))
+        resp = self.client.post("/transaction/commit/{}".format(self.transaction))
         self.transaction = None
 
+        return resp.json()
```

Comment:
should we put this on a try block? I know we don't do this for a majority of the code, but I just ran into that thought. what if the request fails, and resp no longer has the `.json()` method?

Reply:
yes, I believe Noah is right. `__wrap` should handle a failed request and then throwing the exception

Reply:
right, I forgot about it.